src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -22,41 +21,39 @@ import { DefaultValidatingSamCliProcessInvoker } from '../sam/cli/defaultValidat
 import { SamLaunchRequestArgs } from '../sam/debugger/samDebugSession'
 import { RuntimeFamily } from '../../lambda/models/samLambdaRuntime'
 import * as pathutil from '../../shared/utilities/pathUtils'
+import { findParentProjectFile } from '../utilities/workspaceUtils'
 
 export async function getSamProjectDirPathForFile(filepath: string): Promise<string> {
-    const packageJsonPath: string | undefined = await findFileInParentPaths(path.dirname(filepath), 'package.json')
+    const packageJsonPath = await findParentProjectFile(vscode.Uri.parse(filepath), path.join('**', 'package.json'))
     if (!packageJsonPath) {
         throw new Error(`Cannot find package.json for: ${filepath}`)
     }
 
-    return path.dirname(packageJsonPath)
+    return path.dirname(packageJsonPath.fsPath)
 }
 
 /**
  * Applies a full relative path to the Javascript handler that will be stored in the CodeLens commands.
  * @param handlers Handlers to apply relative paths to
- * @param parentDocumentPath Path to the file containing these Lambda Handlers
+ * @param uri URI of the file containing these Lambda Handlers
  */
 export async function decorateHandlerNames(
     handlers: LambdaHandlerCandidate[],
-    parentDocumentPath: string
-): Promise<void> {
-    const parentDir = path.dirname(parentDocumentPath)
-    const packageJsonPath = await findFileInParentPaths(parentDir, 'package.json')
+    uri: vscode.Uri
+): Promise<LambdaHandlerCandidate[]> {
+    const packageJsonFile = await findParentProjectFile(uri, path.join('**', 'package.json'))
 
-    if (!packageJsonPath) {
-        return
+    if (!packageJsonFile) {
+        return []
     }
 
-    const relativePath = getHandlerRelativePath({
-        codeRoot: path.dirname(packageJsonPath),
-        filePath: parentDocumentPath,
-    })
+    const relativePath = path.relative(path.dirname(packageJsonFile.fsPath), path.dirname(uri.fsPath))
 
-    handlers.forEach(handler => {
+    return handlers.map(handler => {
         const handlerName = handler.handlerName
-
         handler.handlerName = normalizeSeparator(path.join(relativePath, handlerName))
```

Comment:
`handlers` is still being mutated, so returning a list from `decorateHandlerNames()` is misleading. Would be better to either avoid mutation or keep the `Promise<void>` return type, which made it clear that the function has side-effects.

Reply:
The goal was to avoid mutation. I'll see what I can do here.

Reply:
Double checking -- was there a change to remove the mutation, or is this currently in progress?

Reply:
So kinda, but I offloaded it to the caller, which in retrospect makes no sense: https://github.com/aws/aws-toolkit-vscode/pull/1049/files#diff-4a09b26abadcd1030a9d7e8172cc714fR202

I'll swap it to here.